hips/tiles/surveys.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from astropy.table import Table
+from collections import OrderedDict
+
+__all__ = [
+    'HipsSurveyList',
+]
+
+
+class HipsSurveyList:
+    """HiPS survey container.
+
+    Parameters
+    ----------
+    surveys : `~astropy.table.Table`
+        HiPS surveys
+    """
+
+    def __init__(self, surveys: Table) -> None:
+        self.surveys = surveys
+
+    @classmethod
+    def read(cls, filename: str) -> 'HipsSurveyList':
+        """Read HiPS list from file (`HipsSurveyList`).
+
+        Parameters
+        ----------
+        filename : str
+            HiPS list filename
+        """
+        with open(filename, 'r', encoding='utf-8', errors='ignore') as fh:
+            text = fh.read()
+
+        return cls.parse(text)
+
+    @classmethod
+    def parse(cls, text: str) -> 'HipsSurveyList':
+        """Parse HiPS list text (`HipsSurveyList`).
+
+        Parameters
+        ----------
+        text : str
+            HiPS list text
+        """
+        surveys = list()
+        print(len(text.split('\n\n')))
+        for raw_survey in text.split('\n\n'):
+            survey = OrderedDict()
+            for line in raw_survey.split('\n'):
```

Comment:
Isn't this duplicating the code we already have in https://hips.readthedocs.io/en/latest/api/hips.HipsDescription.html#hips.HipsDescription.parse
Can you call that instead to do the parsing of each survey?